scripts/overton/EDA_script.py

```diff
@@ -0,0 +1,176 @@
+        file_object.write('0.001:'+ str(ai0001) + ', percentage: ' + str(round(ai0001 / len(df),2)) + '\n')
+        file_object.write('0.0001:'+ str(ai00001) + ', percentage: ' + str(round(ai00001 / len(df),2)) + '\n')
+        file_object.write('ML over: \n')
+        file_object.write('0.1:'+ str(ml01) + ', percentage: ' + str(round(ml01 / len(df),2)) + '\n')
+        file_object.write('0.01:'+ str(ml001) + ', percentage: ' + str(round(ml001 / len(df),2)) + '\n')
+        file_object.write('0.001:'+ str(ml0001) + ', percentage: ' + str(round(ml0001 / len(df),2)) + '\n')
+        file_object.write('0.0001:'+ str(ml00001) + ', percentage: ' + str(round(ml00001 / len(df),2)) + '\n')
+
+    df['n_paragraphs'].hist().write_image(file = 'plots/hist_n_paragraphs.png', format = 'png')
+    df['n_paragraphs'].hist()
+
+    df['n_words'].hist().write_image(file = 'plots/hist_n_words.png', format = 'png')
+    df['n_words'].hist()
+
+    count_language = Counter(spacy_df['paragraph_language'][0])
+    for i in range(1, len(spacy_df)):
+        count_language += Counter(spacy_df['paragraph_language'][i])
+    count_language = eda.count_texts(count_language)
+    count_language.to_excel('excel/count_language.xlsx')
+    count_language
+
+    nouns = []
+    for i in range(len(spacy_df['merged_nouns'])):
+        nouns = np.concatenate((nouns,spacy_df['merged_nouns'][i]))
+    lemmas = []
+    for i in range(len(spacy_df['merged_lemmas'])):
+        lemmas = np.concatenate((lemmas,spacy_df['merged_lemmas'][i]))
+    noun_chunks = []
+    for i in range(len(spacy_df['merged_noun_chunks'])):
+        noun_chunks = np.concatenate((noun_chunks,spacy_df['merged_noun_chunks'][i]))
+
+    with open("EDA_report.txt", "a") as file_object:
+        file_object.write('Number of nouns:'+ str(len(nouns)) + '\n')
+        file_object.write('Number of lemmas:'+ str(len(lemmas)) + '\n')
+        file_object.write('Number of noun_chunks:'+ str(len(noun_chunks)) + '\n')
+
+    print('Number of nouns:', len(nouns))
+    print('Number of lemmas:', len(lemmas))
+    print('Number of noun_chunks:', len(noun_chunks))
+
+    count_nouns = eda.count_texts(nouns)
+    eda.plot_counts(count_nouns, ['obj', 'count']).write_image(file = 'plots/bar_counted_nouns.png', format = 'png')
+    eda.plot_counts(count_nouns, ['obj', 'count'])
+
+    word_counts = Counter(lemmas)
+    wc = WordCloud(width=1600, height=800)
+    wc.generate_from_frequencies(frequencies = word_counts)
+    plt.figure(figsize = (18, 14))
+    plt.imshow(wc)
+    plt.savefig('plots/word_cloud.png')
+
+    count_lemmas= eda.count_texts(lemmas)
+    eda.plot_counts(count_lemmas, ['obj', 'count']).write_image(file = 'plots/bar_counted_lemmas.png', format = 'png')
+    eda.plot_counts(count_lemmas, ['obj', 'count'])
+
+    stopwords = en.Defaults.stop_words
+    top_n_bigrams=eda.get_top_ngram(df['Text_all'], stopwords = stopwords, n = 2, m = 2)[:30]

```

Comment:
zamiast powtarzac 3 razy to samo napisz jakas funkcje pomocnicza na przyszlosc ;)